kong/plugins/http-log/handler.lua

```diff
@@ -72,36 +48,47 @@ local function log(premature, conf, body, name)
   end
   name = "[" .. name .. "] "
 
-  local ok, err
+  local headers
+  local req_url
   local parsed_url = parse_url(conf.http_endpoint)
-  local host = parsed_url.host
-  local port = tonumber(parsed_url.port)
 
-  local sock = ngx.socket.tcp()
-  sock:settimeout(conf.timeout)
-
-  ok, err = sock:connect(host, port)
-  if not ok then
-    ngx.log(ngx.ERR, name .. "failed to connect to " .. host .. ":" .. tostring(port) .. ": ", err)
-    return
+  if parsed_url.userinfo then
+    -- for lua-resty-http client Authorization
+    headers = {
+      ["Content-Type"] = conf.content_type,
+      ["Authorization"] = string_format(
+        "Basic %s", ngx.encode_base64(parsed_url.userinfo))
+    }
+    req_url = parsed_url.scheme .. "://" .. parsed_url.host .. ":" .. parsed_url.port
+      .. parsed_url.path .. (parsed_url.params or "")
+  else
+    headers = {
+      ["Content-Type"] = conf.content_type
+    }
+    req_url = conf.http_endpoint
   end
 
-  if parsed_url.scheme == HTTPS then
-    local _, err = sock:sslhandshake(true, host, false)
-    if err then
-      ngx.log(ngx.ERR, name .. "failed to do SSL handshake with " .. host .. ":" .. tostring(port) .. ": ", err)
-    end
+  local httpc = http.new()
+  httpc:set_timeout(conf.timeout)
+  local res, err = httpc:request_uri(req_url, {
+      ssl_verify = false,
+      method = conf.method:upper(),
+      body = body,
+      headers = headers
+  })
+
+  if not res then
+    ngx.log(ngx.ERR, name .. "request failed: " .. req_url, err)
```

Comment:
Need better formatting. This will not insert a space between `req_url` and `err`. We should also stop using this pattern of Lua-land string concatenations, let's update it while we're at it.
```lua
ngx.log(ngx.ERR, name, "request failed (", req_url, "): ", err)
```